homeassistant/components/notify/xmpp.py

```diff
@@ -31,79 +28,96 @@
     vol.Optional(CONF_TLS, default=True): cv.boolean,
     vol.Optional(CONF_VERIFY, default=True): cv.boolean,
     vol.Optional(CONF_ROOM, default=''): cv.string,
+    vol.Optional(CONF_RESOURCE, default="home-assistant"): cv.string,
 })
 
 
-def get_service(hass, config, discovery_info=None):
+async def async_get_service(hass, config, discovery_info=None):
     """Get the Jabber (XMPP) notification service."""
     return XmppNotificationService(
-        config.get(CONF_SENDER), config.get(CONF_PASSWORD),
-        config.get(CONF_RECIPIENT), config.get(CONF_TLS),
-        config.get(CONF_VERIFY), config.get(CONF_ROOM))
+        config.get(CONF_SENDER), config.get(CONF_RESOURCE),
+        config.get(CONF_PASSWORD), config.get(CONF_RECIPIENT),
+        config.get(CONF_TLS), config.get(CONF_VERIFY),
+        config.get(CONF_ROOM), hass.loop)
 
 
 class XmppNotificationService(BaseNotificationService):
     """Implement the notification service for Jabber (XMPP)."""
 
-    def __init__(self, sender, password, recipient, tls, verify, room):
+    def __init__(self,
+                 sender,
+                 resource,
+                 password,
+                 recipient,
+                 tls,
+                 verify,
+                 room,
+                 loop):
```

Comment:
This form is not very common on home assistant code base

Reply:
two lines then?